lib/Platform/Intl/java/com/facebook/hermes/intl/PlatformNumberFormatterICU.java

```diff
@@ -0,0 +1,358 @@
+            }
+
+            mDecimalFormat.setSignificantDigitsUsed(true);
+        }
+
+        return this;
+    }
+
+    @Override
+    public PlatformNumberFormatterICU configureFractinDigits(IPlatformNumberFormatter.RoundingType roundingType, int minimumFractionDigits, int maximumFractionDigits) {
+        if (roundingType == IPlatformNumberFormatter.RoundingType.FRACTION_DIGITS) {
+            if (minimumFractionDigits >= 0)
+                mDecimalFormat.setMinimumFractionDigits(minimumFractionDigits);
+
+            if (maximumFractionDigits >= 0)
+                mDecimalFormat.setMaximumFractionDigits(maximumFractionDigits);
+
+            mDecimalFormat.setSignificantDigitsUsed(false);
+        }
+
+        return this;
+    }
+
+    @Override
+    public PlatformNumberFormatterICU configureSignDisplay(IPlatformNumberFormatter.SignDisplay signDisplay) {
+        DecimalFormatSymbols symbols = mDecimalFormat.getDecimalFormatSymbols();
+
+        switch (signDisplay) {
+            case NEVER:
+                mDecimalFormat.setPositivePrefix("");
+                mDecimalFormat.setPositiveSuffix("");
+
+                mDecimalFormat.setNegativePrefix("");
+                mDecimalFormat.setNegativeSuffix("");
+
+                break;
+            case ALWAYS:
+            case EXCEPTZERO:
+
+                if (!mDecimalFormat.getNegativePrefix().isEmpty())
+                    mDecimalFormat.setPositivePrefix(new String(new char[]{symbols.getPlusSign()}));
+
+                if (!mDecimalFormat.getNegativeSuffix().isEmpty())
+                    mDecimalFormat.setPositiveSuffix(new String(new char[]{symbols.getPlusSign()}));
+
+                break;
+        }
+        return this;
+    }
+
+    private static MeasureUnit parseUnit(String inUnit) throws JSRangeErrorException {
+
+        // http://unicode.org/reports/tr35/tr35-general.html#Unit_Identifiers
+        // Only core identifiers will be input
+        for (MeasureUnit unit : MeasureUnit.getAvailable()) {
+
+            if (unit.getSubtype().equals(inUnit) || unit.getSubtype().equals(unit.getType() + "-" + inUnit))
+                return unit;
+        }
+
+        throw new JSRangeErrorException("Unknown unit: " + inUnit);
+    }
+
+
+    @Override
+    public PlatformNumberFormatterICU configureUnits(String unit, IPlatformNumberFormatter.UnitDisplay unitDisplay) throws JSRangeErrorException {
+        if (mStyle == Style.UNIT) {
+            MeasureFormat.FormatWidth formatWidth;
+            switch (unitDisplay) {
+                case LONG:
+                    formatWidth = MeasureFormat.FormatWidth.WIDE;
+                    break;
+                case NARROW:
+                    formatWidth = MeasureFormat.FormatWidth.NARROW;
+                    break;
+                case SHORT:
+                default:
+                    formatWidth = MeasureFormat.FormatWidth.SHORT;
+                    break;
+            }
```

Comment:
One of the cool features of java enums is they can have data fields and methods.  So you could store the MeasureFormat.FormatWidth value as data on the enum itself, and replace this switch statement with a simple method call which is just an accessor.  See [an example here](https://gist.github.com/mhorowitz/81a1841036b46ed7fc3ea11306e44ebe) or at https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html.  In a very rudimentary test, this was quite a bit smaller than using a switch to look up a per-enum value.

There are a lot of places where you switch over an enum, where you could instead return a per-enum value.

Reply:
Thanks! I'm heavily leveraging this pattern in the latest iteration ..